x-pack/plugins/observability_solution/inventory/public/hooks/use_discover_redirect.ts

```diff
@@ -7,23 +7,24 @@
 import { useCallback, useMemo } from 'react';
 import type { InventoryEntity } from '../../common/entities';
 import { useAdHocDataView } from './use_adhoc_data_view';
-import { useFetchEntityDefinition } from './use_fetch_entity_definition';
+import { useFetchEntityDefinitionIndexPattern } from './use_fetch_entity_definition_index_patterns';
 import { useKibana } from './use_kibana';
 
 export const useDiscoverRedirect = (entity: InventoryEntity) => {
   const {
     services: { share, application, entityManager },
   } = useKibana();
-  const { entityDefinitions, isEntityDefinitionLoading } = useFetchEntityDefinition(
-    entity.entityDefinitionId
-  );
+  const { entityDefinitionIndexPatterns, isEntityDefinitionIndexPatternsLoading } =
+    useFetchEntityDefinitionIndexPattern(entity.entityType);
 
   const title = useMemo(
     () =>
-      !isEntityDefinitionLoading && entityDefinitions && entityDefinitions?.length > 0
-        ? entityDefinitions[0]?.indexPatterns?.join(',')
+      !isEntityDefinitionIndexPatternsLoading &&
+      entityDefinitionIndexPatterns &&
+      entityDefinitionIndexPatterns?.length > 0
+        ? entityDefinitionIndexPatterns[0].join(',')
```

Comment:
nit: `,` is the default character used in `join`

```suggestion
        ? entityDefinitionIndexPatterns[0].join()
```